xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-notifiers/xwiki-platform-notifications-notifiers-default/src/main/java/org/xwiki/notifications/notifiers/internal/email/AbstractMimeMessageIterator.java

```diff
@@ -364,6 +364,8 @@ public MimeMessage next()
 
             updateFactoryParameters(templateDocumentReference);
             message = this.factory.createMessage(templateDocumentReference, this.factoryParameters);
+            message.setHeader("Auto-Submitted", "auto-generated");
+            message.setHeader("X-Auto-Response-Suppress", "All");
```

Comment:
Great to see this feature exist in the protocol.

However, I'm not sure this is the best location, I would expect something like that to be more somewhere in `createMessage` itself but looks like we don't really have any common code between message factories so that would mean adding it to each of them.

I guess another possibility would be to make it the default in ExtendedMimeMessage constructor, but it might be a bit too much.

WDYT @vmassol ?

Reply:
AFAICS only `Auto-Submitted` is defined in the RFC (https://www.rfc-editor.org/rfc/rfc5436.html#section-2.7.1) the other one is defined by Microsoft according to https://www.arp242.net/autoreply.html. 

Would be good to maybe add some comments to at least explain that. Also reading a bit more the RFC it sounds like the Auto-Submitted field should include the owner-email: 
> The header field "Auto-Submitted: auto-notified" MUST include one or
   both of the following parameters: [owner-email or owner-token]

> I guess another possibility would be to make it the default in ExtendedMimeMessage constructor, but it might be a bit too much.

According to the RFC the header could contain `auto-generated` or `auto-notified`. IMO it makes to always have the header in the emails sent by XWiki, now it's not necessarly always `auto-generated`: it makes sense here for notification, but not necessarily for some other mails (like invitation email for example).